eras/shelley/impl/src/Cardano/Ledger/Shelley/State/CertState.hs

```diff
@@ -63,27 +60,21 @@ shelleyCertPStateL :: Lens' (ShelleyCertState era) (PState era)
 shelleyCertPStateL = lens shelleyCertPState (\ds u -> ds {shelleyCertPState = u})
 {-# INLINE shelleyCertPStateL #-}
 
-shelleyCertVStateL :: Lens' (ShelleyCertState era) (VState era)
-shelleyCertVStateL = lens shelleyCertVState (\ds u -> ds {shelleyCertVState = u})
-{-# INLINE shelleyCertVStateL #-}
-
 toCertStatePairs :: KeyValue e a => ShelleyCertState era -> [a]
-toCertStatePairs certState@(ShelleyCertState _ _ _) =
+toCertStatePairs certState@(ShelleyCertState _ _) =
   let ShelleyCertState {..} = certState
    in [ "dstate" .= shelleyCertDState
       , "pstate" .= shelleyCertPState
-      , "vstate" .= shelleyCertVState
       ]
 
 shelleyObligationCertState :: ShelleyCertState era -> Obligations
-shelleyObligationCertState (ShelleyCertState VState {vsDReps} PState {psDeposits} DState {dsUnified}) =
-  let accum ans drepState = ans <> drepDeposit drepState
-   in Obligations
-        { oblStake = UM.fromCompact (UM.sumDepositUView (UM.RewDepUView dsUnified))
-        , oblPool = F.foldl' (<>) (Coin 0) psDeposits
-        , oblDRep = F.foldl' accum (Coin 0) vsDReps
-        , oblProposal = Coin 0
-        }
+shelleyObligationCertState (ShelleyCertState PState {psDeposits} DState {dsUnified}) =
```

Comment:
If you make it polymorphic and use lenses to access, then you'll be able to reuse it in Conway.
```suggestion
shelleyObligationCertState :: EraCertState era => CertState era -> Obligation
shelleyObligationCertState certState =
```